src/common/net.cc

```diff
@@ -52,12 +52,71 @@ Ipv4::any() {
 
 std::string
 Ipv4::toString() const {
-    static constexpr size_t MaxSize = sizeof("255") * 4 + 3 + 1; /* 4 * 255 + 3 * dot + \0 */
+    
+    // Use the built-in ipv4 string length from arpa/inet.h
+    char buff[INET_ADDRSTRLEN+1];
+    
+    in_addr_t addr;
+    toNetwork(&addr);
+    
+    // Convert the network format address into display format
+    inet_ntop(AF_INET, &addr, buff, INET_ADDRSTRLEN);
+    
+    return std::string(buff);
+}
 
-    char buff[MaxSize];
-    snprintf(buff, MaxSize, "%d.%d.%d.%d", a, b, c, d);
+void Ipv4::toNetwork(in_addr_t *addr) const {
+    // Bitshift the bytes into an in_addr_t (a single 32bit unsigned int);
+    *addr = (uint32_t)(d<<24) | (uint32_t)(c<<16) | (uint32_t)(b<<8) | (uint32_t)a;
+}
 
-    return std::string(buff);
+Ipv6::Ipv6(uint16_t a, uint16_t b, uint16_t c, uint16_t d, uint16_t e, uint16_t f, uint16_t g, uint16_t h)
+    : a(a)
+    , b(b)
+    , c(c)
+    , d(d)
+    , e(e)
+    , f(f)
+    , g(g)
+    , h(h)
+{ }
+
+Ipv6
+Ipv6::any() {
+    return Ipv6(0, 0, 0, 0, 0, 0, 0, 0);
+}
+
+std::string
+Ipv6::toString() const {
+    
+    // Use the built-in ipv6 string length from arpa/inet.h
+    char buff6[INET6_ADDRSTRLEN+1];
+    
+    in6_addr addr;
+    toNetwork(&addr);
+    
+    inet_ntop(AF_INET6, &addr, buff6, INET6_ADDRSTRLEN);
+
+    return std::string(buff6);
+}
+
+void Ipv6::toNetwork(in6_addr *addr6) const {
+    uint16_t temp_ip6[8] = {a, b, c, d, e, f, g, h};
+    uint16_t remap_ip6[8] = {0};
+    uint16_t x, y;
+    
+     // If native endianness is little-endian swap the bytes, otherwise just copy them into the new array
+    if ( htonl(1) != 1 ) {
+        for (uint16_t i = 0; i<8; i++) {
+            x = temp_ip6[i];
+            y = __builtin_bswap16 (x);
```

Comment:
Is there a reason why you did not use "htons()" here?  Under gcc and clang "htons()" might be an alias for "__builtin_bswap16()".  Since this is networking code, "htons()" seems more natural.  Just curious.  Since pistache uses Linux's eventfd, it is unlikely that this code will ever be compiled by something other than gcc or clang (I highly douct that msvc has this builtitn).

Reply:
htons is also fine I'll swap that in to avoid any potential confusion, but AFAIK this is supported in current MSVC (at least that seems to be indicated by a similar discussion on ardupilot).